src/main/java/eu/dissco/dataexporter/repository/DataExporterRepository.java

```diff
@@ -36,18 +39,41 @@ public void addJobToQueue(ExportJob job) throws InvalidRequestException {
         .execute();
   }
 
-  public void markJobAsRunning(UUID id){
+  public Optional<String> getJobResultsIfExists(UUID hashedParams){
+    return context.select(EXPORT_QUEUE.S3_LINK)
+        .from(EXPORT_QUEUE)
+        .where(EXPORT_QUEUE.HASHED_PARAMS.eq(hashedParams))
+        .fetchOptional(EXPORT_QUEUE.S3_LINK);
+  }
+
+  public void updateJobState(UUID id, JobState jobState) {
     context.update(EXPORT_QUEUE)
-        .set(EXPORT_QUEUE.JOB_STATE, JobState.RUNNING)
+        .set(EXPORT_QUEUE.JOB_STATE, jobState)
         .set(EXPORT_QUEUE.TIME_STARTED, Instant.now())
         .where(EXPORT_QUEUE.ID.eq(id))
         .execute();
   }
 
+  public void markJobAsComplete(JobResult jobResult, JobState jobState) {
+    context.update(EXPORT_QUEUE)
+        .set(EXPORT_QUEUE.JOB_STATE, jobState)
+        .set(EXPORT_QUEUE.TIME_COMPLETED, Instant.now())
+        .set(EXPORT_QUEUE.S3_LINK, jobResult.s3Link())
```

Comment:
Starting to doubt if we need to call it S3_Link or just exporter_result or exporter_link to keep it agnostic. We might switched away from AWS then we are stuck with this name